Lib9c/Action/TransferAssets.cs

```diff
@@ -0,0 +1,170 @@
+        {
+            var state = context.PreviousStates;
+            if (context.Rehearsal)
+            {
+                return Map.OrderBy(i => i.Key).Aggregate(state, (current, kv) => current.MarkBalanceChanged(kv.Value.Currency, new[] {Sender, kv.Key}));
+            }
+
+            var addressesHex = GetSignerAndOtherAddressesHex(context, context.Signer);
+            var started = DateTimeOffset.UtcNow;
+            Log.Debug("{AddressesHex}TransferAsset3 exec started", addressesHex);
+
+            state = Map.OrderBy(i => i.Key).Aggregate(state, (current, kv) => Transfer(current, context.Signer, kv.Key, kv.Value));
+            var ended = DateTimeOffset.UtcNow;
+            Log.Debug("{AddressesHex}TransferAsset4 Total Executed Time: {Elapsed}", addressesHex, ended - started);
+
+            return state;
+        }
+
+        public override void LoadPlainValue(IValue plainValue)
+        {
+            var asDict = (Dictionary) plainValue;
+
+            Sender = asDict["sender"].ToAddress();
+            var rawMap = (List)asDict["map"];
+            Map = new Dictionary<Address, FungibleAssetValue>();
+            foreach (var iValue in rawMap)
+            {
+                var list = (List) iValue;
+                var key = list[0].ToAddress();
+                var value = list[1].ToFungibleAssetValue();
+                Map[key] = value;
+            }
+            Memo = asDict.TryGetValue((Text) "memo", out IValue memo) ? memo.ToDotnetString() : null;
+
+            CheckMemoLength(Memo);
+        }
+
+        public void GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+            info.AddValue("serialized", new Codec().Encode(PlainValue));
+        }
+
+        private void CheckMemoLength(string memo)
+        {
+            if (memo?.Length > MemoMaxLength)
+            {
+                string msg = $"The length of the memo, {memo.Length}, " +
+                             $"is overflowed than the max length, {MemoMaxLength}.";
+                throw new MemoLengthOverflowException(msg);
+            }
+        }
+
+        private IAccountStateDelta Transfer(IAccountStateDelta state, Address signer, Address recipient, FungibleAssetValue amount)
+        {
+            if (Sender != signer)
+            {
+                throw new InvalidTransferSignerException(signer, Sender, recipient);
+            }
+
+            if (Sender == recipient)
+            {
+                throw new InvalidTransferRecipientException(Sender, recipient);
+            }
+
+            Address recipientAddress = recipient.Derive(ActivationKey.DeriveKey);
+
+            // Check new type of activation first.
+            // If result of GetState is not null, it is assumed that it has been activated.
+            if (
+                state.GetState(recipientAddress) is null &&
+                state.GetState(Addresses.ActivatedAccount) is Dictionary asDict &&
```

Comment:
It seems that this action will try this `.GetState()` lookup per `Map` count, but they are all also the same. can we move it from this local method to reduce it?